src/h5web/visualizations/shared/models.ts

```diff
@@ -1,8 +1,28 @@
 import { ScaleLinear, ScaleSymLog, scaleSymlog, scaleLinear } from 'd3-scale';
 
+export type Size = { width: number; height: number };
+
 export type Domain = [number, number];
 
-export type Size = { width: number; height: number };
+export interface IndexAxisConfig {
+  indexDomain: Domain;
+  showGrid?: boolean;
+  isLog?: never; // invalid
+}
+
+export interface DataAxisConfig {
+  dataDomain: Domain;
+  showGrid?: boolean;
+  isLog?: boolean;
+}
+
+export type AxisConfig = IndexAxisConfig | DataAxisConfig;
+
+export interface AxisScale {
+  scale: DataScale;
+  scaleFn: DataScaleFn;
+  domain: Domain;
+}
```

Comment:
This is the shape of the objects returned by the scale hooks. As explained, it now includes the domain for convenience.

Reply:
`AxisScale` is becoming redundant: we have `scaleFn` and `domain` which can both derive from `scale`... In fact, we could imagine including `range` instead of `scale` and put the scale creation in the code using `AxisScale`: `scaleFn.domain(domain).range(range)`...

Reply:
Yeah, I did wonder about that, but in practice it's not amazing... TS would probably ask us to keep the domain/range calls on separate lines, we'd run into naming conflicts if destructuring, etc.

I think this will evolve again in the future, but I'm not sure how yet.